sunpy/physics/transforms/differential_rotation.py

```diff
@@ -85,9 +81,360 @@ def diff_rot(ddays, latitude, rot_type='howard', frame_time='sidereal'):
+    # rotation
+    # Start time
+    dstart = parse_time(tstart)
+    dend = parse_time(tend)
+    interval = (dend - dstart).total_seconds() * u.s
+
+    # Get the Sun's position from the vantage point at the start time
+    vstart = kwargs.get("vstart", _calc_P_B0_SD(dstart, spacecraft=spacecraft))
+    # Compute heliographic co-ordinates - returns (longitude, latitude). Points
+    # off the limb are returned as nan
+    longitude, latitude = convert_hpc_hg(x.to(u.arcsec).value,
+                                         y.to(u.arcsec).value,
+                                         b0_deg=vstart["b0"].to(u.deg).value,
+                                         l0_deg=vstart["l0"].to(u.deg).value, 
+                                         dsun_meters=(constants.au * sun.sunearth_distance(t=dstart)).value,
+                                         angle_units='arcsec')
+    longitude = Longitude(longitude, u.deg)
+    latitude = Angle(latitude, u.deg)
+    # Compute the differential rotation
+    drot = diff_rot(interval, latitude, frame_time=frame_time,
+                    rot_type=rot_type)
+
+    # Convert back to heliocentric cartesian in units of arcseconds
+    vend = kwargs.get("vend", _calc_P_B0_SD(dend, spacecraft=spacecraft))
+
+    # It appears that there is a difference in how the SSWIDL function
+    # hel2arcmin and the sunpy function below performs this co-ordinate
+    # transform.
+
+    newx, newy = convert_hg_hpc(longitude.to(u.deg).value + drot.to(u.deg).value,
+                                latitude.to(u.deg).value,
+                                b0_deg=vend["b0"].to(u.deg).value,
+                                l0_deg=vend["l0"].to(u.deg).value,
+                                dsun_meters=(constants.au * sun.sunearth_distance(t=dend)).value,
+                                occultation=False)
+    newx = Angle(newx, u.arcsec)
+    newy = Angle(newy, u.arcsec)
+    return newx.to(u.arcsec), newy.to(u.arcsec)
+
+
+def _calc_P_B0_SD(date, spacecraft=None, arcsec=False):
+    """
+    To calculate the solar P, B0 angles and the semi-diameter.  This
+    function is assigned as being internal as these quantities should be
+    calculated in a part of SunPy that can calculate these quantities
+    accurately.
+
+    Parameters
+    -----------
+    date: `sunpy.time.time`
+        the time at which to calculate the solar P, B0 angles and the
+        semi-diameter.
+
+    spacecraft: { None | "soho" | "stereo_a" | "stereo_b" }
+        calculate the solar P, B0 angles and the semi-diameter from the point
+        of view of either SOHO or either of the STEREO spacecraft.  SOHO sits
+        at the Lagrange L1 point which is about 1% closer to the Sun than the
+        Earth.  Implementation of this seems to require the ability to read
+        SOHO orbit files.
+
+    arcsec: { False | True }
+        return the semi-diameter in arcseconds.
+
+    Returns:
```

Comment:
Much of the functionality was blindly copied from SSW.  I agree - we don't need the arcsec option.  It's crazy!